source/workflowapi/app.py

```diff
@@ -2230,6 +2414,505 @@ def update_workflow_execution_status(id, status, message):
+def create_vocabulary():
+    """ Create an Amazon Transcribe custom vocabulary.
+
+    Body:
+
+        .. code-block:: python
+
+            {
+                'vocabulary_name'='string',
+                'language_code'='af-ZA'|'ar-AE'|'ar-SA'|'cy-GB'|'da-DK'|'de-CH'|'de-DE'|'en-AB'|'en-AU'|'en-GB'|'en-IE'|'en-IN'|'en-US'|'en-WL'|'es-ES'|'es-US'|'fa-IR'|'fr-CA'|'fr-FR'|'ga-IE'|'gd-GB'|'he-IL'|'hi-IN'|'id-ID'|'it-IT'|'ja-JP'|'ko-KR'|'ms-MY'|'nl-NL'|'pt-BR'|'pt-PT'|'ru-RU'|'ta-IN'|'te-IN'|'tr-TR'|'zh-CN',
+                's3uri'='string'
+            }
+
+
+    Returns:
+        This is a proxy for boto3 create_vocabulary and returns the output from that SDK method.
+        See `the boto3 documentation for details <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/transcribe.html#TranscribeService.Client.create_vocabulary>`_
+
+    Raises:
+        See the boto3 documentation for details
+        500: ChaliceViewError - internal server error
+    """
+    # Save the input vocab to a new vocabulary
+    print('create_vocabulary request: '+app.current_request.raw_body.decode())
+    transcribe_client = boto3.client('transcribe', region_name=os.environ['AWS_REGION'])
+    vocabulary_name = json.loads(app.current_request.raw_body.decode())['vocabulary_name']
+    language_code = json.loads(app.current_request.raw_body.decode())['language_code']
+    response = transcribe_client.create_vocabulary(
+        VocabularyName=vocabulary_name,
+        LanguageCode=language_code,
+        VocabularyFileUri=json.loads(app.current_request.raw_body.decode())['s3uri']
+    )
+    return response
+
+
+@app.route('/service/translate/get_terminology', cors=True, methods=['POST'], content_types=['application/json'], authorizer=authorizer)
+def get_terminology():
+    """ Get a link to the CSV formatted description for an Amazon Translate parallel data.
+
+    Body:
+
+    .. code-block:: python
+
+        {
+            'terminology_name'='string'
+        }
+
+    Returns:
+        This is a proxy for boto3 get_terminology and returns the output from that SDK method.
+        See `the boto3 documentation for details <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/translate.html#Translate.Client.get_terminology>`_
+
+    Raises:
+        See the boto3 documentation for details
+        500: ChaliceViewError - internal server error
+    """
+    print('get_terminology request: '+app.current_request.raw_body.decode())
+    translate_client = boto3.client('translate', region_name=os.environ['AWS_REGION'])
+    terminology_name = json.loads(app.current_request.raw_body.decode())['terminology_name']
+    response = translate_client.get_terminology(Name=terminology_name, TerminologyDataFormat='CSV')
+    # Remove response metadata since we don't need it
+    del response['ResponseMetadata']
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Your code is attempting to access the response metadata, `ResponseMetadata`, without performing a `None` check on the response object. This might cause a NoneType TypeError. To prevent this, we recommend that you add a `None` check on the response object before accessing the response metadata.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html)

Reply:
We are not going to address this CodeGuru finding because we want to mirror the official AWS code sample for sigv4 signing: https://docs.aws.amazon.com/code-samples/latest/catalog/python-signv4-v4-signing-get-post.py.html.